spec/Section 3 -- Type System.md

```diff
@@ -962,21 +1424,92 @@ query withNullableVariable($var: String) {
 Note: The Validation section defines providing a nullable variable type to
 a non-null input type as invalid.
 
-**Non-Null type validation**
+**Type Validation**
 
 1. A Non-Null type must not wrap another Non-Null type.
 
 
 ## Directives
 
-A GraphQL schema includes a list of the directives the execution
-engine supports.
+DirectiveDefinition : Description? directive @ Name ArgumentsDefinition? on DirectiveLocations
+
+DirectiveLocations :
+  - `|`? DirectiveLocation
+  - DirectiveLocations | DirectiveLocation
+
+DirectiveLocation : one of
+  `QUERY`                 `SCHEMA`                `ENUM`
+  `MUTATION`              `SCALAR`                `ENUM_VALUE`
```

Comment:
Should we be consistent throughout these names with the `_DEFINITION` suffix? I was confused at first what it meant for a directive to apply to `SCALAR`. I at first thought it meant that something like `query { someField(integerArgument: 5 @mydirective) }` would be valid before I realized it was talking about definitions.

Or maybe could we split them into two groups, `TypeSystemDirectiveLocation` and `OperationalDirectiveLocation` and say that `DirectiveLocation` must be from one of the two groups?

Reply:
Because these are in use, it would be pretty painful to change these values, however I like the idea of grammatically splitting across two groups for clarity

Reply:
Makes sense. 👍 Thanks!!